src/sentry/eventstream/kafka/consumer.py

```diff
@@ -0,0 +1,295 @@
+        if group != synchronize_commit_group:
+            logger.debug('Received consumer offsets update from %r, ignoring...', group)
+            continue
+
+        partition_state_manager.set_remote_offset(topic, partition, offset)
+
+
+def get_earliest_offset(consumer, topic, partition):
+    low, high = consumer.get_watermark_offsets(TopicPartition(topic, partition))
+    return low
+
+
+def get_latest_offset(consumer, topic, partition):
+    low, high = consumer.get_watermark_offsets(TopicPartition(topic, partition))
+    return high
+
+
+class SynchronizedConsumer(object):
+    """
+    This class implements the framework for a consumer that is intended to only
+    consume messages that have already been consumed and committed by members
+    of another consumer group.
+
+    This works similarly to the Kafka built-in ``__consumer_offsets`` topic.
+    The consumer group that is being "followed" (the one that must make
+    progress for our consumer here to make progress, identified by the
+    ``synchronize_commit_group`` constructor parameter/instance attribute) must
+    report its offsets to a topic (identified by the ``commit_log_topic``
+    constructor parameter/instance attribute). This consumer subscribes to both
+    commit log topic, as well as the topic(s) that we are actually interested
+    in consuming messages from. The messages received from the commit log topic
+    control whether or not consumption from partitions belonging to the main
+    topic is paused, resumed, or allowed to continue in its current state
+    without changes.
+
+    The furthest point in any partition that this consumer should ever consume
+    to is the maximum offset that has been recorded to the commit log topic for
+    that partition. If the offsets recorded to that topic move
+    non-monotonically (due to an intentional offset rollback, for instance)
+    this consumer *may* consume up to the highest watermark point. (The
+    implementation here tries to pause consuming from the partition as soon as
+    possible, but this makes no explicit guarantees about that behavior.)
+    """
+    initial_offset_reset_strategies = {
+        'earliest': get_earliest_offset,
+        'latest': get_latest_offset,
+    }
+
+    def __init__(self, bootstrap_servers, consumer_group, commit_log_topic,
+                 synchronize_commit_group, initial_offset_reset='latest', on_commit=None):
+        self.bootstrap_servers = bootstrap_servers
+        self.consumer_group = consumer_group
+        self.commit_log_topic = commit_log_topic
+        self.synchronize_commit_group = synchronize_commit_group
+        self.initial_offset_reset = self.initial_offset_reset_strategies[initial_offset_reset]
+
+        self.__partition_state_manager = SynchronizedPartitionStateManager(
+            self.__on_partition_state_change)
+        self.__commit_log_consumer, self__commit_log_consumer_stop_request = self.__start_commit_log_consumer()
```

Comment:
Missing `.` in `self.__commit_log_consumer_stop_request` ?

Reply:
Eek, good catch.